File: collector/src/main/java/com/bitmovin/analytics/adapters/PlayerAdapter.java
package com.bitmovin.analytics.adapters;

import com.bitmovin.analytics.config.SourceMetadata;
import com.bitmovin.analytics.data.EventData;
import com.bitmovin.analytics.data.manipulators.EventDataManipulatorPipeline;
import com.bitmovin.analytics.features.Feature;
import com.bitmovin.analytics.license.FeatureConfigContainer;
import java.util.Collection;

public interface PlayerAdapter {
    Collection<Feature<FeatureConfigContainer, ?>> init();

    void release();

    void resetSourceRelatedState();

    void registerEventDataManipulators(EventDataManipulatorPipeline pipeline);

    long getPosition();

    Long getDRMDownloadTime();

    void clearValues();

    SourceMetadata getCurrentSourceMetadata();

    EventData createEventData();

    default AdAdapter createAdAdapter() {
        return null;
    }
}

File: collector/src/main/java/com/bitmovin/analytics/adapters/PlayerAdapter.kt
package com.bitmovin.analytics.adapters

import com.bitmovin.analytics.api.DefaultMetadata
import com.bitmovin.analytics.api.SourceMetadata
import com.bitmovin.analytics.data.PlayerInfo
import com.bitmovin.analytics.dtos.EventData
import com.bitmovin.analytics.dtos.FeatureConfigContainer
import com.bitmovin.analytics.features.Feature
import com.bitmovin.analytics.ssai.SsaiService
import com.bitmovin.analytics.stateMachines.PlayerStateMachine

interface PlayerAdapter {
    val stateMachine: PlayerStateMachine
    val position: Long
    val drmDownloadTime: Long?
    var defaultMetadata: DefaultMetadata
    val playerInfo: PlayerInfo
    val isAutoplayEnabled: Boolean?
    val ssaiService: SsaiService

    fun init(): Collection<Feature<FeatureConfigContainer, *>>

    fun release()

    fun resetSourceRelatedState()

    fun createEventData(): EventData

    fun createEventDataForAdSample(): EventData

    fun createEventDataForCustomDataEvent(sourceMetadata: SourceMetadata): EventData

    fun createAdAdapter(): AdAdapter? {
        return null
    }

    fun getCurrentSourceMetadata(): SourceMetadata

    fun triggerLastSampleOfSession()
}
